src/main/java/com/artipie/SliceFromConfig.java

```diff
@@ -0,0 +1,135 @@
+                line, headers, body
+            );
+        } catch (final InterruptedException ex) {
+            Logger.error(this, "Interruption when getting slice from config");
+            throw new IllegalArgumentException(ex);
+        } catch (final ExecutionException ex) {
+            Logger.error(this, "Exception when getting slice from config");
+            throw new IllegalArgumentException(ex);
+        }
+    }
+
+    /**
+     * Find a slice implementation for config.
+     * @param cfg Repository config
+     * @return Slice
+     * @throws ExecutionException If error getting the slice
+     * @throws InterruptedException If error getting the slice
+     */
+    private static Slice build(final RepoConfig cfg) throws InterruptedException,
+        ExecutionException {
+        return cfg.type().thenCombine(
+            cfg.storage(),
+            (type, storage) -> {
+                return new MapOf<String, Function<RepoConfig, Slice>>(
+                    new MapEntry<String, Function<RepoConfig, Slice>>(
+                        "file", config -> new FilesSlice(storage)
+                    ),
+                    new MapEntry<String, Function<RepoConfig, Slice>>(
+                        "npm", config -> new NpmSlice(new Npm(storage), storage)
+                    ),
+                    new MapEntry<String, Function<RepoConfig, Slice>>(
+                        "gem", config -> new GemSlice(storage)
+                    ),
+                    new MapEntry<String, Function<RepoConfig, Slice>>(
+                        "rpm", config -> new RpmSlice(storage)
+                    ),
+                    new MapEntry<String, Function<RepoConfig, Slice>>(
+                        "php",
+                        config -> {
+                            try {
+                                return config.path().thenApply(
+                                    path -> new PhpComposer(path, storage)
+                                ).toCompletableFuture().get();
+                            } catch (final InterruptedException ex) {
+                                Logger.error(
+                                    SliceFromConfig.class, "Interrupted PhpComposer creation"
+                                );
+                                throw new IllegalArgumentException(ex);
+                            } catch (final ExecutionException ex) {
+                                Logger.error(
+                                    SliceFromConfig.class, "Exception getting PhpComposer"
+                                );
+                                throw new IllegalArgumentException(ex);
+                            }
+                        }
+                    ),
+                    new MapEntry<String, Function<RepoConfig, Slice>>(
+                        "maven", config -> new MavenSlice(storage)
+                    )
+                ).get(type).apply(cfg);
```

Comment:
@HDouss we should check if the type exists in the map and throw exception if it doesnt